src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2482,13 +2531,18 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
                     manifestInfo.AddToModuleList(m);
                 }
             }
+
             if (compatiblePSEditions != null)
             {
-                foreach (var psEdition in compatiblePSEditions)
-                {
-                    manifestInfo.AddToCompatiblePSEditions(psEdition);
-                }
+                manifestInfo.AddToCompatiblePSEditions(compatiblePSEditions);
```

Comment:
My thinking is that the `PSModuleInfo` should reflect the manifest. But I think this particular point needs further discussion, because rewriting default logic everywhere is error prone. I'm reluctant to add another public field on to the `PSModuleInfo` (and that would probably merit an RFC anyway).

But so far, the way it works is:
* `PSModuleInfo.CompatiblePSEditions` is exactly what's in the manifest (or an empty array)
* Everywhere we check `CompatiblePSEditions` a default is used if it's null or empty

Reply:
I meant that if `compatiblePSEditions == null` we should add default.

Reply:
Yeah, I'm thinking that we shouldn't.

If we do, then we are not reflecting what the manifest says. So there will be no way to differentiate between a manifest that has `CompatiblePSEditions = @("Desktop")` and no `CompatiblePSEditions` field at all -- all our tools to parse a manifest will incorrectly see them as the same.

For the purposes of this logic, yes we assume a default. But this is quite specific logic. So I think we need to preserve the information for other functionalities that may need to distinguish the scenarios.

But maybe we also need a way to record the default assumed value on the `PSModuleInfo`?

Reply:
I agree with @rjmholt that the member should reflect what is actually in the manifest.